pydantic/exceptions.py

```diff
@@ -11,87 +13,48 @@
 )
 
 
-def type_display(type_: type):
-    try:
-        return type_.__name__
-    except AttributeError:
-        # happens with unions
-        return str(type_)
-
-
-Error = namedtuple('Error', ['exc', 'track', 'index'])
+class Error:
+    def __init__(self, exc: Exception, *, loc: Union[str, int] = None) -> None:
+        self.exc_info = exc
+        self.exc_type = type(exc)
```

Comment:
I guess this isn't necessary as we can call `type(self.exc_info)` when it's needed?

Reply:
yep